# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../blackbox/hotfix.patch
# Copyright (C) 2016 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- blackbox-0.71.1/src/Toolbar.cc.vanilla	2016-12-09 10:03:37.825174435 +0000
+++ blackbox-0.71.1/src/Toolbar.cc	2016-12-09 10:04:20.329175165 +0000
@@ -44,7 +44,7 @@
 {
   timeval now;
   gettimeofday(&now, 0);
-  return (std::max(1000l, ((((resolution - (now.tv_sec % resolution)) * 1000l))
+  return (std::max((time_t)1000, ((((resolution - (now.tv_sec % resolution)) * 1000l))
                            - (now.tv_usec / 1000l))));
 }
 
